### PR TITLE
[Merged by Bors] - Add contribution guidelines

### LINKED ISSUE #21: Add contribution guidelines

### PR DESCRIPTION
**Pull request by lberrymage**
Closes #21

### CODE REVIEW

**Comment by lberrymage** [comment]
@hspil, want to put this through a second set of eyeballs? Do you think anything could be worded better? How's the tone? Is anything confusing or missing? I've written most of this at night, so I expect it to be far from perfect.

P.S. You can preview the markdown with [`python-grip`](https://aur.archlinux.org/packages/python-grip/)

File: CODESTYLE.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+As a general rule, all code and documentation in this repository conforms to the
+[Rust API Guidelines] and [Rust RFC 1574]. We won't check every commit against
+them; that would be impractical. But please at least read through [this
+checklist] to get a feel for what's expected. If you find something in the
+codebase that doesn't conform to the guidelines, feel free to [submit an issue].
+
+## Line wrapping
+
+Line wrapping should be done primarily by `rustfmt` (i.e. don't manually wrap
+long-ish lines). Of course, if your code is significantly harder to read with
```
Comment (hspil) [inline CODESTYLE.md:65]: With? or without?

Comment (lberrymage) [reply CODESTYLE.md:65]: Oh yes, it's supposed to be without. Thank you.

**Pull request closed without merging**